### PR TITLE
 Build site extensions in main build chain 

### PR DESCRIPTION
**Pull request by pakrym**
1. Detect SDK we are running and use it for site extension
1. Delete runtime and host that SDK brings
1. Use `dependencies.props` to get shared runtime version
1. Drop in AspNetCore shared runtime
1. Drop in latest templates

### CODE REVIEW

**Comment by natemcmaster** [comment]
> Drop in AspNetCore shared runtime

This doesn't exist yet when universe is building aspnet/AzureIntegration. This makes me think the site extension targets should be built in aspnet/Universe as this is just another "installer" scenario.

**Comment by pakrym** [comment]
@natemcmaster yes, but there is also a set of functional tests that run against Antares and require site extensions and I'm not sure where to put them if we move site extensions out.

**Comment by natemcmaster** [comment]
Ok. We can leave this here for now, but we should look into making the site extension part of aspnet/Universe. We'll be adding tests to aspnet/Universe for the shared runtime and other installers. This would fall in the same category.

File: build/repo.targets [hunk 1]
```diff
@@ -23,34 +24,69 @@
...
     <UnzipArchive File="%(DotNetCacheArchiveItems.Identity)" Destination="$(SiteExtensionWorkingDirectory)DotNetCache" Condition="'$(DotNetCacheArchives)' != ''" Overwrite="true" />
 
-    <MSBuild Projects="%(SiteExtensions.Project)"
+    <MSBuild Projects="%(SiteExtensions.Idenity)"
       Targets="Restore;Pack"
       Properties="DotnetHomeDirectory=$(SiteExtensionWorkingDirectory);BuildNumber=$(BuildNumber);PackageOutputPath=$(SiteExtensionOutputDirectory)" />
 
   </Target>
 
   <Target Name="PushSiteExtension">
```
Comment (natemcmaster) [inline build/repo.targets]: We should sync up on this. We need to make sure this is publishing to the right places, using the right parameters for PCP.

File: build/repo.targets [hunk 2]
```diff
@@ -23,34 +24,69 @@
...
+
+    <Copy SourceFiles="$(ArtifactDependencyLocation)\%(BundledTemplate.Identity).%(Version).nupkg" DestinationFolder="$(SiteExtensionWorkingDirectory)\sdk\$(_SdkVersion)\Templates" />
+    <Copy SourceFiles="$(ArtifactDependencyLocation)\nuGetPackagesArchive-$(PackageVersion).lzma" DestinationFiles="$(SiteExtensionWorkingDirectory)\sdk\$(_SdkVersion)\nuGetPackagesArchive.lzma" />
+
+    <UnzipArchive File="$(ArtifactDependencyLocation)\aspnetcore-shared-$(PackageVersion)-win7-x86.zip" Destination="$(SiteExtensionWorkingDirectory)" Overwrite="true" />
     <UnzipArchive File="%(RuntimeArchiveItems.Identity)" Destination="$(SiteExtensionWorkingDirectory)" Condition="'$(RuntimeArchives)' != ''" Overwrite="true" />
     <UnzipArchive File="%(DotNetCacheArchiveItems.Identity)" Destination="$(SiteExtensionWorkingDirectory)DotNetCache" Condition="'$(DotNetCacheArchives)' != ''" Overwrite="true" />
 
-    <MSBuild Projects="%(SiteExtensions.Project)"
+    <MSBuild Projects="%(SiteExtensions.Idenity)"
```
Comment (JunTaoLuo) [inline build/repo.targets]: Idenity => Identity?

**Pull request merged**